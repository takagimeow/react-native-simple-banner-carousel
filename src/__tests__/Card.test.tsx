/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import 'react-native';
import { render } from '@testing-library/react-native';

import { Card } from '..';

describe('Card', () => {
  it('Snapshot Test', () => {
    const { toJSON } = render(
      <Card
        id={`banner_id`}
        title="マクドナルド札幌すすきの店"
        description="配送手数料: ￥150・5-15分"
        score={4.7}
        source={{
          uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
        }}
        width={128}
        onPress={() => console.log('hello world')}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
