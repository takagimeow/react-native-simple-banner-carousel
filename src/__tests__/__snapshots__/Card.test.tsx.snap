// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot Test 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fffffff",
        "maxWidth": "100%",
        "paddingBottom": 16,
        "paddingTop": 12,
        "width": 128,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
      }
    }
    style={
      Object {
        "backgroundColor": "#bbb",
        "height": 156,
        "marginBottom": 8,
        "maxWidth": "100%",
        "width": 128,
      }
    }
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 11,
          "fontSize": 14,
        }
      }
    >
      マクドナルド札幌すすきの店
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f0f0f0",
          "borderRadius": 25,
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        4.7
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      配送手数料: ￥150・5-15分
    </Text>
  </View>
</View>
`;
